sklearn/utils/estimator_checks.py

```diff
@@ -73,6 +73,18 @@
                 'RANSACRegressor', 'RadiusNeighborsRegressor',
                 'RandomForestRegressor', 'Ridge', 'RidgeCV']
 ALLOW_NAN = ['Imputer', 'SimpleImputer', 'MICEImputer']
+SKIP_DTYPE_CONVERSION_CHECK = [
+             'AdditiveChi2Sampler', 'BernoulliRBM', 'Binarizer', 'Birch',
+             'CCA', 'DictionaryLearning', 'FactorAnalysis', 'FastICA',
+             'FeatureAgglomeration', 'FunctionTransformer',
+             'GaussianRandomProjection', 'Isomap', 'LatentDirichletAllocation',
+             'LinearDiscriminantAnalysis', 'LocallyLinearEmbedding',
+             'MiniBatchDictionaryLearning', 'MiniBatchSparsePCA', 'NMF',
+             'PLSCanonical', 'PLSRegression', 'PLSSVD', 'RBFSampler',
+             'RandomizedLasso', 'RandomizedLogisticRegression', 'SelectFdr',
+             'SelectFpr', 'SelectFwe', 'SelectKBest', 'SkewedChi2Sampler',
+             'SparsePCA', 'SparseRandomProjection', 'VarianceThreshold']
```

Comment:
We also need to include `BadTransformerWithoutMixin` here and remove the PLS related estimators.